Android/app/src/main/java/com/example/androidgui/UpdateQuanActivity.java

```diff
@@ -0,0 +1,34 @@
+package com.example.androidgui;
+
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+import controller.inventorySystem.InventoryManager;
+
+public class UpdateQuanActivity extends AppCompatActivity implements View.OnClickListener{
+    Button btn;
+    EditText name;
+    EditText usage;
+    InventoryManager im = new InventoryManager();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_update_freshness);
+        btn = findViewById(R.id.button9);
+        name = findViewById(R.id.textInputEditText9);
+        usage = findViewById(R.id.textInputEditText10);
+        btn.setOnClickListener(this);
+    }
+    @Override
+    public void onClick(View v) {
+        String iname = name.getText().toString();
+        String iusage = usage.getText().toString();
+        String message = im.newQuantity(iname,iusage);
```

Comment:
The controller should not directly be returning output to the presenter. Provide information through the output boundary.

(e.g. 1. Presenter calls on method in controller
2. controller calls on method in use case
3. use case calls the output boundary to run a method in the presenter and set the value in the presenter)